File: datasets/rs-keep-code_davinci_001_temp_0.2/HumanEval_92_any_int.rs
/// Create a function that takes 3 numbers.
/// Returns true if one of the numbers is equal to the sum of the other two, and all numbers are integers.
/// Returns false in any other cases.
/// Examples
/// any_int(5, 2, 7) ➞ True
/// any_int(3, 2, 2) ➞ False
/// any_int(3, -2, 1) ➞ True
/// any_int(3.6, -2.2, 2) ➞ False
fn any_int(x: f64, y: f64, z: f64) -> bool {
    let sum = x + y + z;
    let x_int = x.floor() == x;
    let y_int = y.floor() == y;
    let z_int = z.floor() == z;
    let sum_int = sum.floor() == sum;
    let x_eq_sum = x == sum;
    let y_eq_sum = y == sum;
    let z_eq_sum = z == sum;
    if x_int && y_int && z_int && sum_int {
        if x_eq_sum || y_eq_sum || z_eq_sum {
            return true;
        }
    }
    false

}

fn main() {
    let candidate = any_int;
    assert_eq!(candidate(2.0, 3.0, 1.0), true);
    assert_eq!(candidate(2.5, 2.0, 3.0), false);
    assert_eq!(candidate(1.5, 5.0, 3.5), false);
    assert_eq!(candidate(2.0, 6.0, 2.0), false);
    assert_eq!(candidate(4.0, 2.0, 2.0), true);
    assert_eq!(candidate(2.2, 2.2, 2.2), false);
    assert_eq!(candidate(-4.0, 6.0, 2.0), true);
    assert_eq!(candidate(2.0, 1.0, 1.0), true);
    assert_eq!(candidate(3.0, 4.0, 7.0), true);
    assert_eq!(candidate(3.0, 4.0, 7.0), false);
}

File: datasets/py-keep-code_davinci_001_temp_0.2/HumanEval_92_any_int.py
def any_int(x: float, y: float, z: float) -> bool:
    """
    Create a function that takes 3 numbers.
    Returns true if one of the numbers is equal to the sum of the other two, and all numbers are integers.
    Returns false in any other cases.
    
    Examples
    any_int(5, 2, 7) ➞ True
    
    any_int(3, 2, 2) ➞ False

    any_int(3, -2, 1) ➞ True
    
    any_int(3.6, -2.2, 2) ➞ False
  

    
    """
    if x.is_integer() and y.is_integer() and z.is_integer():
        if x + y == z or x + z == y or y + z == x:
            return True
        else:
            return False
    else:
        return False


def check(candidate):
    assert candidate(2, 3, 1) == True
    assert candidate(2.5, 2, 3) == False
    assert candidate(1.5, 5, 3.5) == False
    assert candidate(2, 6, 2) == False
    assert candidate(4, 2, 2) == True
    assert candidate(2.2, 2.2, 2.2) == False
    assert candidate(-4, 6, 2) == True
    assert candidate(2, 1, 1) == True
    assert candidate(3, 4, 7) == True
    assert candidate(3.0, 4, 7) == False

def test_check():
    check(any_int)

test_check()
